planet/specs.py

```diff
@@ -111,14 +110,18 @@ def get_item_types(product_bundle=None):
     '''If given product bundle, get specific item types supported by Orders
     API. Otherwise, get all item types supported by Orders API.'''
     spec = _get_product_bundle_spec()
-    if product_bundle:
-        item_types = spec['bundles'][product_bundle]['assets'].keys()
-    else:
-        product_bundle = get_product_bundles()
-        all_item_types = []
-        for bundle in product_bundle:
-            all_item_types += [*spec['bundles'][bundle]['assets'].keys()]
-        item_types = set(all_item_types)
+
+    try:
+        if product_bundle:
+            item_types = set(spec['bundles'][product_bundle]['assets'].keys())
+        else:
+            item_types = set(
+                itertools.chain.from_iterable(
+                    spec['bundles'][bundle]['assets'].keys()
+                    for bundle in get_product_bundles()))
+    except KeyError:
+        raise validate_bundle(product_bundle)
```

Comment:
@kevinlacaille I think validating the product bundle should be done before line 116. Then we would probably not see a KeyError ever except if the bundle object had no 'assets' item.

Reply:
You're right, that makes more sense. Done!